Objects/unicodeobject.c

```diff
@@ -2217,6 +2221,20 @@ PyUnicode_FromWideChar(const wchar_t *u, Py_ssize_t size)
     if (size == 0)
         _Py_RETURN_UNICODE_EMPTY();
 
+#ifdef HAVE_NON_UNICODE_WCHAR_T_REPRESENTATION
+    /* Oracle Solaris uses non-Unicode internal wchar_t form for
+       non-Unicode locales and hence needs conversion to UCS-4 first. */
+    if (_Py_LocaleUsesNonUnicodeWchar()) {
+        wchar_t* converted = _Py_ConvertWCharFormToUCS4(u, size);
+        if (!converted) {
+            return NULL;
+        }
```

Comment:
The code relies on the fact that wchar_t is 32-bit, please add an assertion somewhere:

```
Py_BUILD_ASSERT(sizeof(wchar_t) == 4);
```

Reply:
Yes, I placed it into `_Py_ConvertWCharForm` which is probably the best place.

Reply:
Added to _Py_ConvertWCharForm, thanks.